test/apps/credentials/test.ts

```diff
@@ -11,35 +11,79 @@ describe('credentials', function() {
 	before('build app', () => build({ cwd: __dirname }));
 	before('start runner', async () => {
 		r = await new AppRunner().start(__dirname);
+		r.page.setExtraHTTPHeaders({
+			authorization: 'my-auth'
+		});
 	});
 
 	after(() => r && r.end());
 
-	// tests
+	// cookie tests
 	it('sends cookies when using this.fetch with credentials: "include"', async () => {
-		await r.load('/credentials?creds=include');
+		await r.load('/credentials?creds=include&type=cookie');
 
 		assert.equal(
 			await r.text('h1'),
 			'a: 1, b: 2, max-age: undefined'
 		);
 	});
 
-	it('does not send cookies when using this.fetch without credentials', async () => {
-		await r.load('/credentials');
+	it('does not send cookies when using this.fetch with credentials: "omit"', async () => {
+		await r.load('/credentials?creds=omit&type=cookie');
 
 		assert.equal(
 			await r.text('h1'),
 			'unauthorized'
 		);
 	});
 
+	it('sends cookies to same origin when using this.fetch without credentials', async () => {
+		await r.load('/credentials?type=cookie');
+
+		assert.equal(
+			await r.text('h1'),
+			'a: 1, b: 2, max-age: undefined'
+		);
+	});
+
+	// TODO: does not send cookies to different origin with credentials: "same-origin"
```

Comment:
This comment (and the other TODO below) are making me nervous. We definitely do not want to be sending cookies or authorization headers to external servers during SSR by default. This isn't an acceptable known limitation to be addressed later on.

The code in `get_page_handler` doesn't look like it would do that, but I haven't actually tested it. This behavior would need to be fixed (if it is indeed broken), and the tests would need to be updated.

Reply:
I think the TODO is a bit poorly worded. My understanding of the TODO is that it was a TODO to add a test to assert that it does not do that. This PR doesn't actually send credentials to other origins unless the user specifically sets `opts.credentials === 'include'`

I think that the best course of action here is to just drop these TODOs while merging. I'm not sure that adding such tests are necessary for a couple reasons. First is that the current code does not have such tests. Second is that Sapper's tests are integration tests and it doesn't seem very straightforward to setup a another server on another origin and assert that it receives a request without credentials. Though if you do think there's some change that would be useful here, I'm happy to open up a new version of this PR with those changes.

Reply:
Thanks. No I don't think better tests should block this PR. I appreciate the sanity check and the confirmation that we're not leaving a big security hole as a TODO. The updating of the tests I was asking for can be as simple as fixing the comments.